### PR TITLE
Stop root path possibility from metadata filename

### PR DESCRIPTION
**Pull request by gdesmar**
Leaving the possibility for a user to provide a full rooted path using the --outfile option on the commandline.

### CODE REVIEW

File: cart/cart.py [hunk 1]
```diff
@@ -604,7 +604,7 @@ def main():
                             backup_name = backup_name[:-5]
                         else:
                             backup_name += ".uncart"
-                        output_file = cur_metadata.get("name", backup_name)
+                        output_file = cur_metadata.get("name", backup_name).lstrip("/")
```
Comment (cccs-rs) [inline cart/cart.py:607]: How would this play out on other OS like Windows? Maybe we should strip/replace any path separators?

**Comment by malwarefrank** [comment]
I recommend using os.path functions or pathlib.Path objects instead of trying to parse the strings yourself.  For example:

```python
def make_rel(p: str):
    """
    Given a path string.
    Remove the root drive and path separator, to turn it from absolute to relative. 
    """
    # if the path is not absolute
    if not os.path.isabs(p):
        # return it unmodified
        return p
    # otherwise, remove drive if present
    drive, rest = os.path.splitdrive(p)
    # remove both Windows and Posix root separator
    return rest.lstrip("\\/")
```

**Comment by gdesmar** [comment]
> I recommend using os.path functions or pathlib.Path objects instead of trying to parse the strings yourself. For example:
> 
> ```python
> def make_rel(p: str):
>     ...
> ```

I'll look into rewriting it using more os.path or pathlib.Path utilities, but the goal is to stop directory traversal attacks that are possible on os.path.join. Here are two examples:
A simple directory traversal attack
`open(os.path.join("/tmp/", make_rel("./../.././///./../../etc/passwd")),'r').read()`
A directory traversal attack that starts with a letter, knowing a folder in which the user is currently extracting the cart file (i.e. expecting them to be in their home directory)
`open(os.path.join("/home/al", make_rel(".ssh/../../../.././../../etc/passwd")), 'r').read()`
You could imagine, in the case of the cart utility, being able to overwrite /etc/passwd with the content of the cart file.

The other solution is to error-out if we detect a "./" in the path, but I wanted to give a way to still get the file.

**Comment by cccs-sgaron** [comment]
@malwarefrank We went over the code and did some test and will use a much simpler method that will make use of os.path functions...

**Review by cccs-sgaron — APPROVED** [review]

**Pull request merged**